app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,21 @@
+class PostsController < ApplicationController
+  def create
+    post = current_user.profile.posts.build(:body => params[:post][:body])
```

Comment:
I see here that you build the post in the posts create controller.

I actually render the view with an instance variable that has a new post and use that new post in a creation form.